From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Peter=20Hosta=C4=8Dn=C3=BD?= <hostacny.peter@gmail.com>
Date: Sat, 26 Sep 2026 10:39:32 +0200
Subject: [PATCH] Update
 Jellyfin.Server/Migrations/Routines/20260915104305_FixNullEncoderPreset.cs

Co-authored-by: theguymadmax <theguymadmax@proton.me>
(cherry picked from commit b9cfd4e3456b04b792dd7e7684552821477c5e19)
---
 .../Routines/20260915104305_FixNullEncoderPreset.cs           | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Jellyfin.Server/Migrations/Routines/20260915104305_FixNullEncoderPreset.cs b/Jellyfin.Server/Migrations/Routines/20260915104305_FixNullEncoderPreset.cs
index 489f9e7255..f8e63f1920 100644
--- a/Jellyfin.Server/Migrations/Routines/20260915104305_FixNullEncoderPreset.cs
+++ b/Jellyfin.Server/Migrations/Routines/20260915104305_FixNullEncoderPreset.cs
@@ -63,12 +63,12 @@ internal class FixNullEncoderPreset : IAsyncMigrationRoutine
         }
 
         var nilAttribute = encoderPreset.Attribute(_xsiNamespace + "nil");
-        if (nilAttribute is null || !string.Equals(nilAttribute.Value, "true", StringComparison.OrdinalIgnoreCase))
+        if (!string.Equals(nilAttribute?.Value, "true", StringComparison.OrdinalIgnoreCase) && !string.IsNullOrEmpty(encoderPreset.Value))
         {
             return Task.CompletedTask;
         }
 
-        nilAttribute.Remove();
+        nilAttribute?.Remove();
         encoderPreset.Value = nameof(EncoderPreset.auto);
         document.Save(path, SaveOptions.DisableFormatting);
         _logger.LogInformation("Replaced null encoder preset with auto in encoding configuration");
